src/libcadet/model/parts/TwoDimensionalConvectionDispersionOperator.cpp

```diff
@@ -872,30 +874,22 @@ bool TwoDimensionalConvectionDispersionOperator::configure(UnitOpIdx unitOpIdx,
 bool TwoDimensionalConvectionDispersionOperator::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
 	bool hasChanged = false;
+	std::vector<int> dir_old = _dir;
 
 	if (!_velocity.empty())
 	{
 		// _curVelocity has already been set to the network flow rate in setFlowRates()
 		// the direction of the flow (i.e., sign of _curVelocity) is given by _velocity
-
-		active const* const dirs = getSectionDependentSlice(_velocity, _nRad, secIdx);
-		if (secIdx > 0)
-		{
-			active const* const dirsPrev = getSectionDependentSlice(_velocity, _nRad, secIdx - 1);
-			for (unsigned int i = 0; i < _nRad; ++i)
-			{
-				if (dirs[i] * dirsPrev[i] < 0.0)
-				{
-					hasChanged = true;
-					break;
-				}
-			}
-		}
+		active const* const dir_new = getSectionDependentSlice(_velocity, _nRad, secIdx);
 
 		for (unsigned int i = 0; i < _nRad; ++i)
 		{
-			if (dirs[i] < 0.0)
+			_dir[i] = (dir_new[i] > 0);
+			if (_dir[i] * dir_old[i] < 0.0)
+			{
+				hasChanged = true;
 				_curVelocity[i] *= -1.0;
+			}
```

Comment:
Doing it this way allows to get rid of the local `dir_new` variable:
```cpp
const int newDir = (dir_new[i] >= 0.0) ? 1 : -1;
if (_dir[i] * newDir < 0)
{
	hasChanged = true;
	_curVelocity[i] *= -1.0;
}
_dir[i] = newDir;
```

Reply:
> Doing it this way allows to get rid of the local `dir_new` variable:

here you mean `dir_old`, right?